service/flags.go

```diff
@@ -70,10 +86,40 @@ func flags() *flag.FlagSet {
 		"feature-gates",
 		"Comma-delimited list of feature gate identifiers. Prefix with '-' to disable the feature. '+' or no prefix will enable the feature.")
 
+	flagSet.BoolVar(&BuildFlag, buildInfoFlag, false,
+		"Displays list of components available in collector distribution in yaml format",
+	)
+
 	return flagSet
 }
 
 func getConfigFlag(flagSet *flag.FlagSet) []string {
 	cfv := flagSet.Lookup(configFlag).Value.(*configFlagValue)
 	return append(cfv.values, cfv.sets...)
 }
+
+func getBuildInfo(set CollectorSettings) (error, []byte) {
+	components := componentsOutput{}
+	for ext, _ := range set.Factories.Extensions {
+		components.Extensions = append(components.Extensions, ext)
+	}
+	for prs, _ := range set.Factories.Processors {
+		components.Processors = append(components.Processors, prs)
+	}
+	for rcv, _ := range set.Factories.Receivers {
+		components.Receivers = append(components.Receivers, rcv)
+	}
+	for exp, _ := range set.Factories.Exporters {
+		components.Exporters = append(components.Exporters, exp)
+	}
+	components.Version = set.BuildInfo.Version
+
+	yamlData, err := yaml.Marshal(components)
+
```

Comment:
Remove this empty line, keeping the error check close to where the error is received.

Reply:
Ok, thank you. On it.